ls1intum/ArTEMiS
# Feature/sa improvements1

### Checklist

### Motivation and Context


We want to improve the UI, the code quality and achieve more consistency.

### Description

- refactored the way on how to display the question text (edit, participate, statistics)
- refactored the html and css file
- changed that when submittedText is not completely correct the background changes to yellow
- replaced Levenshtein library with fuzzySearch library
- added markdown when displaying edit, participate and statistic
- spots break correctly when the windows gets smaller
- multiple words can be selected on visual mode (for spot creation)
### Steps for Testing


1. Create SA quizzes
2. Checked if markdown is included for edit (visual mode), participate and statistic
3. check that multiple words can be selected in visual mode also when using markdown
4. Did all 4 workflows (Edit, Participate, Statistic and Reevaluate) for 2 questions.


### Screenshots


New css layout:

Edit:

![image](https://user-images.githubusercontent.com/41108487/55379411-516cad00-551d-11e9-8330-6a48cdef2113.png)

![image](https://user-images.githubusercontent.com/41108487/55379552-bfb16f80-551d-11e9-8079-1eaa6082ca5d.png)



Participate:

![image](https://user-images.githubusercontent.com/41108487/55379368-2bdfa380-551d-11e9-9ddd-ec50ed58795d.png)
![image](https://user-images.githubusercontent.com/41108487/55379387-3a2dbf80-551d-11e9-8f06-aea307894467.png)

![image](https://user-images.githubusercontent.com/41108487/55380508-b249b480-5520-11e9-9b98-047f9df60d7a.png)

![image](https://user-images.githubusercontent.com/41108487/55380519-b7a6ff00-5520-11e9-9109-17ab489da564.png)




Statistics:

![image](https://user-images.githubusercontent.com/41108487/55379238-b1168880-551c-11e9-9eba-53c44f4c02d1.png)

![image](https://user-images.githubusercontent.com/41108487/55379333-094d8a80-551d-11e9-816f-c246e30e5961.png)

![image](https://user-images.githubusercontent.com/41108487/55380544-c8577500-5520-11e9-8fbb-3e5ace6d0a50.png)



Screenshots with old css layout:
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/41108487/55270228-c4b3bc00-529c-11e9-92a5-45d9e356759b.png">

<img width="1144" alt="image" src="https://user-images.githubusercontent.com/41108487/55270230-d72df580-529c-11e9-9873-c5a003f76085.png">

<img width="885" alt="image" src="https://user-images.githubusercontent.com/41108487/55270347-05f89b80-529e-11e9-8a47-a0041a9606b4.png">

<img width="893" alt="image" src="https://user-images.githubusercontent.com/41108487/55270354-1446b780-529e-11e9-9e12-b4da8e4eb7e4.png">

![video](https://user-images.githubusercontent.com/41108487/55272320-31d64a00-52bb-11e9-971f-8c82972b27e1.gif)

![video2](https://user-images.githubusercontent.com/41108487/55272324-3f8bcf80-52bb-11e9-8c48-a12ad1fb49b1.gif)

![video](https://user-images.githubusercontent.com/41108487/55272359-c771d980-52bb-11e9-84c1-58050565019c.gif)

![video2](https://user-images.githubusercontent.com/41108487/55272361-cd67ba80-52bb-11e9-801a-78d3b2211b9e.gif)
